### PR TITLE
Move okbuck task to its own gradle task class.

### PR DESCRIPTION
**Pull request by runningcode**

### CODE REVIEW

File: buildSrc/src/main/java/com/uber/okbuck/core/task/OkBuckTask.java [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+import java.util.stream.Collectors;
+
+import static com.uber.okbuck.OkBuckGradlePlugin.OKBUCK_DEFS;
+
+public class OkBuckTask extends DefaultTask {
+
+  @Input public OkBuckExtension okBuckExtension;
+
+  @TaskAction
+  void doOkBuckTask() {
```
Comment (kageiit) [inline buildSrc/src/main/java/com/uber/okbuck/core/task/OkBuckTask.java]: can you rename this method to `okbuck`

Comment (runningcode) [inline buildSrc/src/main/java/com/uber/okbuck/core/task/OkBuckTask.java]: done!

File: buildSrc/src/main/java/com/uber/okbuck/core/task/OkBuckTask.java [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+            kotlinDeps != null ? kotlinDeps.getLeft(): null,
+            kotlinDeps != null ? kotlinDeps.getRight(): null);
+  }
+
+  @Override public String getGroup() {
+    return OkBuckGradlePlugin.GROUP;
+  }
+
+  @Override public String getDescription() {
+    return "Setup okbuck cache and dependencies";
```
Comment (kageiit) [inline buildSrc/src/main/java/com/uber/okbuck/core/task/OkBuckTask.java]: This should be handled by the `setupOkbuck` Task. root okbuck task description should be more like "okbuck task for the root project"

File: buildSrc/src/main/groovy/com/uber/okbuck/OkBuckGradlePlugin.groovy [hunk 3]
```diff
@@ -99,11 +93,11 @@ class OkBuckGradlePlugin implements Plugin<Project> {
         // Create tasks
         Task setupOkbuck = project.task('setupOkbuck')
         setupOkbuck.setGroup(GROUP)
-        setupOkbuck.setDescription("Setup okbuck cache and dependencies")
+        setupOkbuck.setDescription()
 
-        Task okBuck = project.task(OKBUCK)
-        okBuck.setGroup(GROUP)
-        okBuck.setDescription("Generate BUCK files")
+        Task okBuck = project.tasks.create(OKBUCK, OkBuckTask, {
```
Comment (kageiit) [inline buildSrc/src/main/groovy/com/uber/okbuck/OkBuckGradlePlugin.groovy]: okbuck task is not safe to be an incremental task. You should set it as never up to date (which `doLast` does as a side effect). When it is truly incremental, we can lift that restriction

File: buildSrc/src/main/groovy/com/uber/okbuck/OkBuckGradlePlugin.groovy [hunk 4]
```diff
@@ -99,11 +93,11 @@ class OkBuckGradlePlugin implements Plugin<Project> {
         // Create tasks
         Task setupOkbuck = project.task('setupOkbuck')
         setupOkbuck.setGroup(GROUP)
-        setupOkbuck.setDescription("Setup okbuck cache and dependencies")
```
Comment (kageiit) [inline buildSrc/src/main/groovy/com/uber/okbuck/OkBuckGradlePlugin.groovy]: Please undo this

File: buildSrc/src/main/java/com/uber/okbuck/core/task/OkBuckTask.java [hunk 5]
```diff
@@ -0,0 +1,107 @@
...
+  }
+
+  @InputFile
+  public File okbuckDefs() {
+    return getProject().file(OKBUCK_DEFS);
```
Comment (kageiit) [inline buildSrc/src/main/java/com/uber/okbuck/core/task/OkBuckTask.java]: This should be an output file?

File: buildSrc/src/main/java/com/uber/okbuck/core/task/OkBuckTask.java [hunk 6]
```diff
@@ -0,0 +1,107 @@
...
+  }
+
+  @OutputFile
+  public File dotBuckConfig() {
+    return getProject().file(".buckconfig");
```
Comment (kageiit) [inline buildSrc/src/main/java/com/uber/okbuck/core/task/OkBuckTask.java]: This is only generated if it does not exist so I am not sure we should call this an outputfile. We can check if we can remove the generation of this file entirely as the contents are empty anyways

File: buildSrc/src/main/groovy/com/uber/okbuck/OkBuckGradlePlugin.groovy (hunk 4 above)
Comment (runningcode) [inline buildSrc/src/main/groovy/com/uber/okbuck/OkBuckGradlePlugin.groovy]: whoops, i copied the wrong description

File: buildSrc/src/main/java/com/uber/okbuck/core/task/OkBuckTask.java (hunk 2 above)
Comment (runningcode) [inline buildSrc/src/main/java/com/uber/okbuck/core/task/OkBuckTask.java]: updated!

File: buildSrc/src/main/java/com/uber/okbuck/core/task/OkBuckTask.java (hunk 5 above)
Comment (runningcode) [inline buildSrc/src/main/java/com/uber/okbuck/core/task/OkBuckTask.java]: yup

File: buildSrc/src/main/groovy/com/uber/okbuck/OkBuckGradlePlugin.groovy (hunk 3 above)
Comment (runningcode) [inline buildSrc/src/main/groovy/com/uber/okbuck/OkBuckGradlePlugin.groovy]: ok, i've added this to the constructor to ensure that the up to date check always returns false
`getOutputs().upToDateWhen(Specs.satisfyNone());`

File: buildSrc/src/main/java/com/uber/okbuck/core/task/OkBuckTask.java (hunk 6 above)
Comment (runningcode) [inline buildSrc/src/main/java/com/uber/okbuck/core/task/OkBuckTask.java]: ah, i've removed the `@OutputFile` annotation. do you think it should be an `@InputFile`?
ive disabled the up to date check so i think it shouldn't matter.

**Pull request merged**